tsl/src/deparse.c

```diff
@@ -395,6 +395,9 @@ deparse_get_tabledef(TableInfo *table_info)
 	deparse_columns(create_table, rel);
 
 	appendStringInfoChar(create_table, ')');
+#if PG12_GE
+	appendStringInfo(create_table, " USING \"%s\" ", get_am_name(rel->rd_rel->relam));
+#endif
```

Comment:
Right now, our policies say that the PG version on the data nodes should be the same or higher than the one on the access node, so that should not cause problems, but have you considered if there is a scenario where this can fail?

Reply:
Thats a good point, but I do not think this is the place where it should be checked. As far as I remember we do the checking on data node connection.

Reply:
I'm more considering if explicitly adding the `USING` clause even when not given to the original `CREATE TABLE` statement can cause any issues.

For example, if the default is different on the access node and the data node and we force this to be the AM of the access node.

I think it is OK to commit without any changes to the code, because I can't imagine a case where it could be a problem, but you might have some idea.

Reply:
I understand your point. It just feels that adding a check against default will be more error prone in the end, if we send `USING` only in some cases.

Reply:
FWIW, I think the access node should decide the "default" AM so I think it is a feature rather than a bug to always include the USING clause.